ha/core/system_health/system_health.py

```diff
@@ -402,15 +403,16 @@ def process_event(self, healthevent: HealthEvent):
                     # If health is already stored and its a node_health, check further
                     stored_genration_id = current_health_dict["events"][0]["specific_info"]["generation_id"]
                     incoming_generation_id = healthevent.specific_info["generation_id"]
-                    incoming_health_status = current_health_dict["events"][0]["status"]
+                    incoming_health_status = healthevent.event_type
                     pod_restart_val = current_health_dict["events"][0]["specific_info"]["pod_restart"]
                     # Update the current health value itself.
                     latest_health = EntityHealth.read(current_health)
-                    if stored_genration_id != incoming_generation_id:
-                        if incoming_health_status == status:
-                            # If the generation id matches and stored node health matches
-                            # with incoming node health, means online event received first
-                            # instead of failed event in delete scenario
+                    if (component_type == "node") and (stored_genration_id != incoming_generation_id):
```

Comment:
I am not sure. But can you check if any constant is defined for this "node" in system_health directory. And if yes, can you use it?
If no, can you check if you can add it in the const file and use!?

Reply:
Done.

Reply:
Thanks